### ISSUE TITLE
Add RoomManager methods for querying the system.

### ISSUE THREAD

**Issue by icecoldtofu** [godotengine/godot-proposals#3082]
### Describe the project you are working on

A typical FPS using the new Rooms and Portals.

### Describe the problem or limitation you are having in your project

I would like to use `MultiMeshInstance` for spawning decals (like bullet holes and explosion marks) on surfaces in the game. 

A `MultiMeshInstance` is culled on the basis of AABB encompassing all of its instances. This means one `MultiMeshInstance` per map, instancing meshes at player-controlled locations would still probably draw all of them all of the time.

Instead I would like to have one "decals" `MultiMeshInstance` as a child of every `Room` so I could add a decal to the one that is in the room where the bullet impact happened. But to do that I need to know which room is it.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Having a method like `RoomManager.room_at(point: Vector3) -> Room` would make this possible.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I think some of the room finding logic from `RoomManager::_autoplace_object` could be broken out into its own method and exposed to GDScript. Exposing `Room::contains_point` would also be nice.

It could be used like this:
```
var rm: RoomManager
var hit: Vector3
var decal_mm: MultiMesh

var room = rm.room_at(hit)

if room != null:
    decal_mm = room.get_node("decals").multimesh
    decal_mm.set_instance_transform(decal_mm.visible_instance_count, Transform(hit))
    decal_mm.visible_instance_count += 1
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This can be worked around in GDScript by recursively walking the roomlist after the conversion, collecting the convex hull of every room into a list. Testing the point against every plane in every convex shape in the list, on every bullet hit. Room priority resolution would also be necessary.

### Is there a reason why this should be core and not an add-on in the asset library?

Performance.

**Comment by Calinou** [comment]
cc @lawnjelly

**Comment by lawnjelly** [comment]
Yup we should be able to add that. In fact I did have a bunch of this type functionality in LPortal (my original module I wrote), I agree it can be very useful. :+1:
